server/app.py

```diff
@@ -73,6 +73,7 @@ def main():
     tornado.options.define("port", default=8000, help="host port", type=int)
     tornado.options.define("server_host", default="localhost", help="host name", type=str)
     tornado.options.define("server_port", default=8090, help="host port", type=int)
+    tornado.options.define("public_url", default="ws://localhost:8090/ws_json", help="public server url", type=str)
```

Comment:
Not needed to make it as option. Declare var and set its value with help server host and port options

Reply:
Your suggestion will not work. server host:port is not necessarily a public URL - you can host your system on the server, python backend and sc-server would connect through localhost, for example, but sc-web frontend has to know an endpoint of sc-server that will be accessible through the internet (localhost will not work, because it's client's localhost, not server's localhost). Ports and hostnames of public_url may and usually will differ, so it's important to have an option for that.

Reply:
As I understood from conversation with @FallenChromium about that, we really need that flag if we are running server and web on separate machines and this url's might differ in that kind of situation. Let him correct me, if I'm mistaken.

Reply:
Okay. You are right!